File: Python/Produk.py
# Komentarnya sama aja sebenernya kaya di yang lain (mager nulis hehehe)

class Produk:
    def __init__(self, nama: str = "", merek: str = "", harga: int = 0, stok: int = 0):
        self.nama = str(nama)
        self.merek = str(merek)
        self.harga = int(harga)
        self.stok = int(stok)

    def setNama(self, nama: str) -> None:
        self.nama = str(nama)

    def getNama(self) -> str:
        return self.nama

    def setMerek(self, merek: str) -> None:
        self.merek = str(merek)

    def getMerek(self) -> str:
        return self.merek

    def setHarga(self, harga: int) -> None:
        self.harga = int(harga)

    def getHarga(self) -> int:
        return self.harga

    def setStok(self, stok: int) -> None:
        self.stok = int(stok)

    def getStok(self) -> int:
        return self.stok

    def print(self) -> None:
        print("Nama:", self.nama)
        print("Merek:", self.merek)
        print("Harga:", self.harga)
        print("Stok:", self.stok)

File: PHP/Produk.php
<?php

class Produk {
    public $nama;
    public $merek;
    public $harga;
    public $stok;
    public $gambarPath;

    public function __construct($nama = "", $merek = "", $harga = 0, $stok = 0, $gambarPath = "") {
        $this->nama = (string)$nama;
        $this->merek = (string)$merek;
        $this->harga = (int)$harga;
        $this->stok = (int)$stok;
        $this->gambarPath = (string)$gambarPath;
    }

    public function toArray() {
        return [
            'nama' => $this->nama,
            'merek' => $this->merek,
            'harga' => $this->harga,
            'stok' => $this->stok,
            'gambarPath' => $this->gambarPath
        ];
    }

    public static function fromArray($arr) {
        return new Produk(
            isset($arr['nama']) ? $arr['nama'] : "",
            isset($arr['merek']) ? $arr['merek'] : "",
            isset($arr['harga']) ? $arr['harga'] : 0,
            isset($arr['stok']) ? $arr['stok'] : 0,
            isset($arr['gambarPath']) ? $arr['gambarPath'] : ""
        );
    }
}
